src/service.rs

```diff
@@ -0,0 +1,273 @@
+    use windows_service::{
+        service_manager::{ServiceManager, ServiceManagerAccess},
+        service::ServiceAccess,
+    };
+
+    let manager_access = ServiceManagerAccess::CONNECT;
+    let code = match ServiceManager::local_computer(None::<&str>, manager_access) {
+        Ok(service_manager) => {
+            let service_name = &service.name;
+            let access = ServiceAccess::QUERY_CONFIG | ServiceAccess::QUERY_STATUS;
+            match service_manager.open_service(service_name, access) {
+                Ok(service) => {
+                    let maybe_config = match service.query_config() {
+                        Ok(config) => Some(config),
+                        Err(error) => {
+                            error!(message = "Error retrieving service config.", %error);
+                            None
+                        },
+                    };
+
+                    let maybe_status = match service.query_status() {
+                        Ok(status) => Some(status),
+                        Err(error) => {
+                            error!(message = "Error retrieving service status.", %error);
+                            None
+                        },
+                    };
+
+                    let name_string = format!("{:?}", service_name);
+
+                    let state_string = match maybe_status {
+                        Some(status) => format!("{:?}", status.current_state),
+                        None => "Unknown".to_string(),
+                    };
+
+                    let state_width = state_string.len() + 2;
+                    let name_width = name_string.len() + 2;
+
+                    println!("{0: <state_width$} {1: <name_width$} Display Name", "State", "Name", state_width = state_width, name_width = name_width);
+                    println!("{0:-<state_width$} {1:-<name_width$} -------------", "-", "-", state_width = state_width, name_width = name_width);
+                    match maybe_config {
+                        Some(config) => {
+                            println!("{0: <state_width$} {1: <name_width$} {2:#?}", state_string, name_string, config.display_name, state_width = state_width, name_width = name_width);
+                            exitcode::OK
+                        }
+                        _ => exitcode::SOFTWARE
+                    }
+
+                },
+                Err(error) => {
+                    error!(message = "Error opening service.", %error);
+                    exitcode::SOFTWARE
+                }
+            }
+        },
+        Err(error) => {
+            error!(message = "Error connecting to service manager.", %error);
+            exitcode::SOFTWARE
+        }
+    };
+
+    code
+}
+
+#[cfg(windows)]
+fn control_service(service: &ServiceInfo, action: ControlAction) -> exitcode::ExitCode {
+    use crate::vector_windows;
+
+    if action == ControlAction::Status {
```

Comment:
I don't think  this is idiomatic Rust. I'm open to suggestions on how to improve this code

Reply:
I think this is totally ok. :) You might want to use a `match` statement though and only catch the commands you need here.